src/benchmarks/gc/src/exec/env/make_memory_load.c

```diff
@@ -329,38 +377,68 @@ int main(const int argc, char** argv)
     Mem mem = init_mem(args);
 
     // Initial adjust may take a while as it writes to all the pages.
-    // Just to be sure, adjust twice with a sleep in between.
+    // Therefore, do various attempts to get the right load going.
     int attempts = 0;
+
     while (TRUE) {
-        AdjustKind ak = adjust(&mem);
+
+#if VERBOSE
+        printf("\nITERATION NUMBER: %d\n", attempts + 1);
+#endif
+
+        // If by the last attempt to get the desired memory load we have an
+        // acceptable difference (+-5%), then allow the test to continue with a
+        // message warning the user of this delta.
+        //
+        // NOTE: DO NOT add commas to any fprintf messages. Run_single_test.py
+        // expects a very specific format depending on the AdjustKind result here.
+        AdjustKind ak = adjust(&mem, attempts);
         if (ak == adjust_kind_no_adjust)
         {
-            // No need to adjust
+            // No need to adjust as we got the memory load we required.
+            // Do NOT change this message, as run_single_test.py is expecting it
+            // to know everything went fine here.
+            fprintf(stderr, "make_memory_load finished starting up\n");
             break;
         }
-        else if (attempts == 5)
+        else if (ak == adjust_kind_within_threshold)
         {
+            // No need to adjust as we are within the acceptable memory load threshold.
```

Comment:
>// No need to ad [](start = 12, length = 16)

Should we attempt to get closer to the target if we still have attempts left? 
It is more likely that we will get closer to the target than that we get out of the threshold range, so seems like a good gamble to try.

Reply:
This case can only occur in such a case. The function `adjust()` can only return `adjust_kind_within_threshold` if we are at the last attempt.

Reply:
This only occurs in such a case. The `adjust()` function can only return `adjust_kind_within_threshold` if we are at the last attempt.